File: packages/typescript/test/package.test.ts
import { readFileSync } from "node:fs";

import { describe, expect, it } from "vitest";

import { DOMAIN, VERSION } from "../src/index.js";

const pkg = JSON.parse(
  readFileSync(new URL("../package.json", import.meta.url), "utf-8"),
) as { version: string };

describe("quanttide-work 包元数据", () => {
  it("领域英文名", () => {
    expect(DOMAIN).toBe("knowledge-work");
  });

  it("包版本与 package.json 一致", () => {
    expect(VERSION).toBe(pkg.version);
  });
});

File: packages/go/pkg/package_test.go
package quanttide_work_test

import (
	"testing"

	quanttide_work "github.com/quanttide/quanttide-work-toolkit/packages/go/pkg"
)

func TestDomain(t *testing.T) {
	if quanttide_work.Domain != "knowledge-work" {
		t.Fatalf("领域英文名不符：%s", quanttide_work.Domain)
	}
}

func TestVersion(t *testing.T) {
	if quanttide_work.Version != "0.1.0" {
		t.Fatalf("包版本不符：%s", quanttide_work.Version)
	}
}
